internal/states/resource.go

```diff
@@ -24,12 +24,12 @@ type Resource struct {
 	// but only the ones of InstanceKeyType are considered current.
 	Instances map[addrs.InstanceKey]*ResourceInstance
 
-	// ProviderConfig is the absolute address for the provider configuration that
+	// ProviderInstance is the absolute address for the provider configuration that
 	// most recently managed this resource. This is used to connect a resource
 	// with a provider configuration when the resource configuration block is
 	// not available, such as if it has been removed from configuration
 	// altogether.
-	ProviderConfig addrs.AbsProviderConfig
+	ProviderInstance addrs.AbsProviderConfig
```

Comment:
This change is incorrect.  The instance data is stored separately in the Instances field above.